spec/03-plugins/25-oauth2/03-access_spec.lua

```diff
@@ -4067,6 +4067,85 @@ describe("Plugin: oauth2 [#" .. strategy .. "]", function()
       }, json)
     end)
   end)
+
+  describe("Plugin: oauth2 anonymous with username #" .. strategy, function()
+    local proxy_client
+    local admin_client
+    local anonymous
+
+    lazy_setup(function()
+      anonymous = assert(admin_api.consumers:insert {
+        username = "test",
+      })
+
+      local service = assert(admin_api.services:insert({
+        url = "https://mockbin.org"
+      }))
+
+      local route = assert(admin_api.routes:insert({
+        hosts      = { "anonymous-with-username.com" },
+        protocols  = { "http", "https" },
+        service    = service
+      }))
+
+      assert(admin_api.oauth2_plugins:insert({
+        route = { id = route.id },
+        config = {
+          anonymous = "test",
+        },
+      }))
+    end)
+
+    before_each(function()
+      proxy_client = helpers.proxy_client()
+      admin_client = helpers.admin_client()
+    end)
+
+    after_each(function()
+      if proxy_client then proxy_client:close() end
+      if admin_client then admin_client:close() end
+    end)
+
+    it("consumer cache consistency", function()
+      helpers.pwait_until(function()
+        print("retrieving consumer: ", anonymous.usernam)
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/consumers/" .. anonymous.username,
+        })
+        print(require("inspect")(res))
+
+        local res = assert(proxy_client:send {
+          method = "GET",
+          path = "/request",
+          headers = {
+            ["Host"] = "anonymous-with-username.com",
+          }
+        })
+        assert.response(res).has.status(200)
+        local body = assert.response(res).has.jsonbody()
+        assert.are.equal("true", body.headers["x-anonymous-consumer"])
```

Comment:
Should we use `assert.equal`?

Reply:
Maybe both are fine? I see a lot of usages of `assert.are.equal`